app/main.py

```diff
@@ -1,2 +1,159 @@
+from fractions import Fraction
+from typing import Any, Hashable, Iterable, Optional
+
+
 class Dictionary:
-    pass
+    class __Node:
+        def __init__(
+                self,
+                key: Optional[Hashable] = None,
```

Comment:
I think these params shouldn't be optional, if you refactor hash table to hold None values instead of empty Nodes it will be possible